tailnet/tunnel.go

```diff
@@ -91,6 +92,41 @@ func (a AgentCoordinateeAuth) Authorize(req *proto.CoordinateRequest) error {
 	return nil
 }
 
+type ClientUserCoordinateeAuth struct {
+	Database UpdateQuerier
+}
+
+func (a ClientUserCoordinateeAuth) Authorize(ctx context.Context, req *proto.CoordinateRequest) error {
+	if tun := req.GetAddTunnel(); tun != nil {
+		uid, err := uuid.FromBytes(tun.Id)
+		if err != nil {
+			return xerrors.Errorf("parse add tunnel id: %w", err)
+		}
+		err = a.Database.AuthorizeTunnel(ctx, uid)
+		if err != nil {
+			return xerrors.Errorf("workspace agent not found or you do not have permission")
+		}
+	}
```

Comment:
Would be nice if we didn't repeat the code from the rest of this function, which is identical to `ClientCoordinateeAuth`